types/DecimalLit.hpp

```diff
@@ -0,0 +1,293 @@
+   * @param other Other DecimalLit to be compared.
+   * @return True if this is less than or equal to other,
+   *         false otherwise.
+   **/
+  inline bool operator<=(const DecimalLit& other) const {
+    return data_ <= other.data_;
+  }
+
+  /**
+   * @brief Operator overloading for "greater than or equal to".
+   *
+   * @param other Other DecimalLit to be compared.
+   * @return True if this is greater than or equal to other,
+   *         false otherwise.
+   **/
+  inline bool operator>=(const DecimalLit& other) const {
+    return data_ >= other.data_;
+  }
+
+  /**
+   * @brief Operator overloading for "equal to".
+   *
+   * @param other Other DecimalLit to be compared.
+   * @return True if this is equal to other,
+   *         false otherwise.
+   **/
+  inline bool operator==(const DecimalLit& other) const {
+    return data_ == other.data_;
+  }
+
+  /**
+   * @brief Operator overloading for "not equal to".
+   *
+   * @param other Other DecimalLit to be compared.
+   * @return True if this is not equal to other,
+   *         false otherwise.
+   **/
+  inline bool operator!=(const DecimalLit& other) const {
+    return data_ != other.data_;
+  }
+
+  /**
+   * @brief Operator overloading for "negate".
+   *
+   * @return Negative of this.
+   **/
+  inline DecimalLit operator-() const {
+    DecimalLit result;
+    result.data_ = -result.data_;
+    return result;
+  }
+
+  /**
+   * @brief Operator overloading for "plus".
+   *
+   * @param other Other DecimalLit to be added.
+   * @return Sum of this and other.
+   **/
+  inline DecimalLit operator+(const DecimalLit& other) const {
+    DecimalLit result;
+    result.data_ = data_ + other.data_;
+    return result;
+  }
+
+  /**
+   * @brief Operator overloading for "subtract".
+   *
+   * @param other Other DecimalLit to be subtract.
+   * @return Subtraction of other from this.
+   **/
+  inline DecimalLit operator-(const DecimalLit& other) const {
+    DecimalLit result;
+    result.data_ = data_ - other.data_;
+    return result;
+  }
+
+  /**
+   * @brief Operator overloading for "multiply".
+   *
+   * @param other Other DecimalLit to be multiplied.
+   * @return Multiplication of this and other.
+   **/
+  inline DecimalLit operator*(const DecimalLit& other) const {
+    DecimalLit result;
+    result.data_ = (data_ * other.data_) / kMaxFractionInt;
+    return result;
+  }
+
+  /**
+   * @brief Operator overloading for "division".
+   *
+   * @param other Divisor DecimalLit.
+   * @return Division of this with other.
+   **/
+  inline DecimalLit operator/(const DecimalLit& other) const {
+    DecimalLit result;
+    result.data_ = (data_ * kMaxFractionInt) / other.data_;
+    return result;
+  }
+
+  /**
+   * @brief Operator overloading for "modulo".
+   *
+   * @param other Mod DecimalLit.
+   * @return This modulo other.
+   **/
+  inline DecimalLit operator%(const DecimalLit& other) const {
+    DecimalLit result;
+    result.data_ = data_ % other.data_;
```

Comment:
This doesn't look right to me (I might be mistaken). For a Decimal like 10.59, I don't think % should be defined at all. The semantics of this operation (as defined) aren't clear to me.